resource-aggregate/cqrs/eventstore/eventstore.go

```diff
@@ -13,16 +13,28 @@ type VersionQuery struct {
 
 // SnapshotQuery used to load events from snapshot.
 type SnapshotQuery struct {
-	GroupID           string //filter by group ID
-	AggregateID       string //filter to certain aggregateID, groupID is required
-	SnapshotEventType string //required
+	GroupID     string //filter by group ID
+	AggregateID string //filter to certain aggregateID, groupID is required
 }
 
+type SaveStatus int
+
+const (
+	OK                   SaveStatus = 0  // events were stored
+	ConcurrencyException SaveStatus = 1  // need to reload events from store, because the versions of events are obsolete by concurrent store.
+	NeedSnapshot         SaveStatus = 2  // the first event need to be snapshot, because store for aggregate is exhausted.
+	FAIL                 SaveStatus = -1 // error occurs
+)
+
 // EventStore provides interface over eventstore. More aggregates can be grouped by groupID,
 // but aggregateID of aggregates must be unique against whole DB.
 type EventStore interface {
-	Save(ctx context.Context, groupID string, aggregateID string, events []Event) (concurrencyException bool, err error)
-	SaveSnapshot(ctx context.Context, groupID string, aggregateID string, event Event) (concurrencyException bool, err error)
+	// Save save events to path.
+	// All events must have same AggregateID and GroupID.
+	// All events must have filled AggregateID, GroupID and Eventype.
+	// Versions among events must continues ascending.
+	// Only first event can be a snapshot.
+	Save(ctx context.Context, events ...Event) (status SaveStatus, err error)
```

Comment:
Why do we support more events? What's the use case to save more of them, when it happens? The command is atomic operation; the second command cannot be executed in case one is pending so I cannot imagine the situation when this will in our design happen.

Reply:
Yes we want. In future we can store snapshot,event directly. but it is more complicated.

Reply:
Why not now, why in the future?